File: yggdrasil/examples/gs_lesson3/src/gs_lesson3.R
# Import library for input/output
library(reticulate)
library(zeallot)
ygg <- import('yggdrasil.interface.YggInterface')

# Initialize input/output channels
in_channel <- ygg$YggInput('input')
out_channel <- ygg$YggOutput('output')

# Loop until there is no longer input or the queues are closed
while(TRUE) {

  # Receive input from input channel
  # If there is an error, the flag will be False
  c(flag, msg) %<-% in_channel$recv()
  if (!flag) {
    print('No more input.')
    break
  }

  # Print received message
  print(msg)

  # Send output to output channel
  # If there is an error, the flag will be False
  flag = out_channel$send(msg)
  if (!flag) {
    print('Error sending output.')
    break
  }

}

File: yggdrasil/examples/gs_lesson3/src/gs_lesson3.jl
# Import library for input/output channels
using Yggdrasil

# Initialize input/output channels
in_channel = Yggdrasil.YggInterface("YggInput", "input")
out_channel = Yggdrasil.YggInterface("YggOutput", "output")

# Loop until there is no longer input or the queues are closed
while true

    # Receive input from input channel
    # If there is an error, the flag will be false
    flag, msg = in_channel.recv()
    if (!flag)
        println("No more input.")
	break
    end

    # Print received message
    println(msg)

    # Send output to output channel
    # If there is an error, the flag will be false
    flag = out_channel.send(msg)
    if (!flag)
        println("Error sending output.")
        break
    end

end  # while